sdk/cosmos/azure-spring-data-cosmos/src/main/java/com/azure/spring/data/cosmos/CosmosFactory.java

```diff
@@ -26,70 +21,73 @@
 
     private static final Logger LOGGER = LoggerFactory.getLogger(CosmosFactory.class);
 
-    private final CosmosConfig config;
+    private final CosmosAsyncClient cosmosAsyncClient;
 
-    private static final boolean IS_TELEMETRY_ALLOWED =
-        PropertyLoader.isApplicationTelemetryAllowed();
+    private final String databaseName;
 
     private static final String USER_AGENT_SUFFIX =
         Constants.USER_AGENT_SUFFIX + PropertyLoader.getProjectVersion();
 
-    private String getUserAgentSuffix() {
+    private static String getUserAgentSuffix() {
         String suffix = ";" + USER_AGENT_SUFFIX;
-
-        if (IS_TELEMETRY_ALLOWED || config.isAllowTelemetry()) {
-            suffix += ";" + MacAddress.getHashMac();
-        }
-
         return suffix;
```

Comment:
We can make this inline. 
```suggestion
        return ";" + USER_AGENT_SUFFIX;
```